tests/NLog.Web.Tests/FakeHostEnvironment.cs

```diff
@@ -0,0 +1,88 @@
+﻿#if ASP_NET_CORE
+#if ASP_NET_CORE2
+using IHostEnvironment = Microsoft.AspNetCore.Hosting.IHostingEnvironment;
+using Microsoft.Extensions.FileProviders;
+#endif
+#if ASP_NET_CORE3
+using Microsoft.Extensions.Hosting;
+#endif
+using Microsoft.Extensions.FileProviders;
+using NLog.Web.DependencyInjection;
+#else
+using NLog.Web.Internal;
+#endif
+
+namespace NLog.Web.Tests
+{
+    /// <summary>
+    /// Faked implementation of IHostEnvironment designed for unit testing.
+    /// </summary>
+    public class FakeHostEnvironment : IHostEnvironment
+    {
+#if ASP_NET_CORE
```

Comment:
Can the the mock be moved into the NLog.Web-project alone, and be stripped from ASP_NET_CORE-details ?